benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -770,10 +663,155 @@ void GraphicsBenchmarkApp::DrawExtraInfo()
     ImGui::NextColumn();
 }
 
+void GraphicsBenchmarkApp::RenderScene(grfx::SwapchainPtr swapchain, PerFrame& frame, uint32_t imageIndex)
+{
+    grfx::RenderPassPtr currentRenderPass = GetRenderpass(swapchain, imageIndex);
+
+    // Snapshot some scene-related values for the current frame
+    uint32_t currentSphereCount   = pSphereInstanceCount->GetValue();
+    uint32_t currentDrawCallCount = pDrawCallCount->GetValue();
+
+    frame.cmd->TransitionImageLayout(currentRenderPass->GetRenderTargetImage(0), PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_PRESENT, grfx::RESOURCE_STATE_RENDER_TARGET);
+    frame.cmd->BeginRenderPass(currentRenderPass);
+    {
+        frame.cmd->SetScissors(GetScissor());
+        frame.cmd->SetViewports(GetViewport());
+
+        // Draw SkyBox
+        frame.cmd->BindGraphicsPipeline(mSkyBox.pipeline);
+        frame.cmd->BindIndexBuffer(mSkyBox.mesh);
+        frame.cmd->BindVertexBuffers(mSkyBox.mesh);
+        {
+            SkyboxData data;
+            data.MVP = mCamera.GetViewProjectionMatrix() * glm::scale(float3(500.0f, 500.0f, 500.0f));
+            mSkyBox.uniformBuffer->CopyFromSource(sizeof(data), &data);
+
+            frame.cmd->PushGraphicsUniformBuffer(mSkyBox.pipelineInterface, /* binding = */ 0, /* set = */ 0, /* bufferOffset = */ 0, mSkyBox.uniformBuffer);
+            frame.cmd->PushGraphicsSampledImage(mSkyBox.pipelineInterface, /* binding = */ 1, /* set = */ 0, mSkyBoxTexture.sampledImageView);
+            frame.cmd->PushGraphicsSampler(mSkyBox.pipelineInterface, /* binding = */ 2, /* set = */ 0, mSkyBoxTexture.sampler);
+        }
+        frame.cmd->DrawIndexed(mSkyBox.mesh->GetIndexCount());
+
+        // Draw sphere instances
+        const size_t pipelineIndex = mGraphicsPipelinesIndexer.GetIndex({pKnobVs->GetIndex(), pKnobPs->GetIndex(), pKnobVbFormat->GetIndex(), pKnobVertexAttrLayout->GetIndex()});
+        frame.cmd->BindGraphicsPipeline(mPipelines[pipelineIndex]);
+        const size_t meshIndex = mMeshesIndexer.GetIndex({pKnobLOD->GetIndex(), pKnobVbFormat->GetIndex(), pKnobVertexAttrLayout->GetIndex()});
+        frame.cmd->BindIndexBuffer(mSphereMeshes[meshIndex]);
+        frame.cmd->BindVertexBuffers(mSphereMeshes[meshIndex]);
+        {
+            uint32_t mSphereIndexCount  = mSphereMeshes[meshIndex]->GetIndexCount() / kMaxSphereInstanceCount;
+            uint32_t indicesPerDrawCall = (currentSphereCount * mSphereIndexCount) / currentDrawCallCount;
+            // Make `indicesPerDrawCall` multiple of 3 given that each consecutive three vertices (3*i + 0, 3*i + 1, 3*i + 2)
+            // defines a single triangle primitive (PRIMITIVE_TOPOLOGY_TRIANGLE_LIST).
+            indicesPerDrawCall -= indicesPerDrawCall % 3;
+            for (uint32_t i = 0; i < currentDrawCallCount; i++) {
+                SphereData data                 = {};
```

Comment:
Could this be moved out of the for loop?

Reply:
Done